### ISSUE TITLE
CoreML export crashes on Apple Silicon (M4) during mlprogram serialization : coremltools 9.0 + torch 2.11

### ISSUE THREAD

**Issue by amanharshx** [ultralytics/ultralytics#24306]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

hi, been playing around with CoreML exports lately and hitting a weird crash

setup:
  - macOS Apple Silicon (M4), Python 3.11.9
  - torch 2.11.0, coremltools 9.0, ultralytics 8.4.41

trying to export like this:
  
`model.export(format="coreml", imgsz=640, dynamic=True, half=True, int8=False, nms=True, batch=1)`

i know ->  `nms=True and dynamic=True` can't both be true at the same time for CoreML, so i expected something to get overridden. but i didn't expect a full crash.

everything runs fine -> all the MIL passes hit 100% - but then it crashes right after backend_mlprogram pipeline finishes, during what looks like the final save step:

Running MIL backend_mlprogram pipeline: 100%|...| 12/12
python3(3096,0x1f18c2c40) malloc: *** error for object 0x600000001: pointer being freed was not allocated
[1] 3096 abort      python3 temp.py

interesting thing - i exported the same model on the Ultralytics Platform with the same args and it worked. inspected the metadata of that model and found:
- platform ran torch==2.11.0+cpu on Linux (not Apple Silicon)
- nms got overridden to False automatically (end2end model)
- batch stayed at 1, platform did NOT change it to 16

so the conversion itself seems fine, crash is only happening locally on Apple Silicon during serialization. is this a known coremltools 9.0 + torch 2.11 issue on Apple Silicon? any working version combo or workaround?

### Additional

This is my setup: 

`ultralytics==8.4.41`
`torch==2.11.0`
`coremltools==9.0`
`Python 3.11.9`
`macOS Apple Silicon M4`

**Comment by UltralyticsAssistant** [comment]
👋 Hello @amanharshx, thank you for reporting this CoreML export issue on Apple Silicon 🚀 This is an automated response to help gather the right debugging details, and an Ultralytics engineer will also assist you soon.
We recommend a quick visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and answers to common questions.
Since this appears to be a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) so we can reproduce the crash reliably. Helpful details would include:
- the exact export script
- the exact model used
- the full terminal output/logs
- the output of `yolo checks`
- whether the same crash occurs after upgrading to the latest package versions
If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
What model is this?

**Comment by amanharshx** [comment]
@Y-T-G YOLO26n

**Comment by Y-T-G** [comment]
It's probably related to numpy version.

You can enable `faulthandler` to get the crash log

**Comment by amanharshx** [comment]
@Y-T-G numpy is within the required range for coremltools (>=1.14.5,<=2.3.5):

`$ python -c "import numpy; print(numpy.__version__)"`
`2.3.5`

---

here's the faulthandler output --->

crash is in _get_proxy_and_spec inside coremltools/models/model.py, not numpy:

Fatal Python error: Aborted

Current thread 0x00000001f18c2c40 (most recent call first):
    File ".../coremltools/models/model.py", line 617 in _get_proxy_and_spec
    File ".../coremltools/models/model.py", line 532 in __init__
    File ".../coremltools/converters/mil/converter.py", line 237 in _mil_convert
    File ".../coremltools/converters/mil/converter.py", line 186 in mil_convert
    File ".../coremltools/converters/_converters_entry.py", line 646 in convert
    File ".../ultralytics/utils/export/coreml.py", line 204 in torch2coreml

all MIL passes complete successfully (100%), crash happens when coremltools tries to initialize the CoreML proxy - looks like an ANE validation step on Apple Silicon. is there a way to skip proxy loading or force CPU-only during conversion?

**Comment by amanharshx** [comment]
i fixed with this : 

skip_model_load=True in ct.convert() is probably skipping the _MLModelProxy / ANE init that crashes on Apple Silicon.

:)

**Comment by glenn-jocher** [comment]
Thanks for confirming — that points to an upstream `coremltools` Apple Silicon proxy-init bug rather than `YOLO26n` itself; if you’re up for it, please open a small PR against [`ultralytics/utils/export/coreml.py`](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/utils/export/coreml.py) to pass `skip_model_load=True` in `ct.convert()`, and also report it upstream to [coremltools](https://github.com/apple/coremltools) with your minimal repro so it can be fixed at the source.
